public/app/core/components/ThresholdManager/ThresholdManager.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { react2AngularDirective } from 'app/core/utils/react2angular';
+import { ThresholdHandler } from './ThresholdHandler';
+
+export interface IProps {
+  thresholds: any[];
+  className?: any;
+  yPos: (v: any) => any;
+  yPosInvert: (v: any) => any;
+  onChange: (v: any, i: number) => any;
+}
+
+export class ThresholdManager extends React.Component<IProps, any> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: 0,
+    };
+
+    this.onChange = this.onChange.bind(this);
```

Comment:
instead of binding event handlers like this you can just declare the functions like this, 
```javascript
 onChange = (index, newValue) => {
    this.props.onChange(newValue, index);
}
```
example
https://github.com/grafana/grafana/blob/master/public/app/containers/AlertRuleList/AlertRuleList.tsx#L30

Reply:
Yes, arrow function should solve this issue.